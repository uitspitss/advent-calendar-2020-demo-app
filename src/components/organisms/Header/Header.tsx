import React, { FC, useState } from 'react';
import tw, { styled } from 'twin.macro';
import Link from 'next/link';

import { MenuToggleButton } from 'components/atoms/MenuToggleButton';
import { HeaderMenu } from 'components/molecules/HeaderMenu';

export type HeaderProps = {
  children?: never;
  className?: string;
};

type Props = {
  isOpen: boolean;
  handleClick: () => void;
} & HeaderProps;

const Component: FC<Props> = (props) => {
  const { className, isOpen, handleClick } = props;

  return (
    <header className={className}>
      <div>
        <nav>
          <div className="title">
            <Link href="/">
              <span>*WEBSITE TITLE*</span>
            </Link>
          </div>
          <div className="toggleButton">
            <MenuToggleButton
              isOpen={isOpen}
              handleClick={handleClick}
              type="button"
            />
          </div>
          <div className="menuList">
            <HeaderMenu />
          </div>
        </nav>
      </div>
    </header>
  );
};

const StyledComponent = styled(Component)<Pick<Props, 'isOpen'>>`
  & > div {
    ${tw`bg-primary-300 mb-2 lg:mb-4`}
    & > nav {
      ${tw`flex flex-wrap items-center justify-between py-2 px-2 lg:px-6`}

      & > .title {
        ${tw`flex items-center flex-shrink-0 mr-6`}
        & > span {
          ${tw`text-gray-100 font-semibold text-xl tracking-tight`}
        }
      }

      & > .toggleButton {
        ${tw`flex lg:hidden`}
      }

      & > .menuList {
        ${tw`w-full flex-grow lg:flex lg:items-center lg:w-auto`}
        ${({ isOpen }) => (isOpen ? tw`block` : tw`hidden lg:block`)}
      }
    }
  }
`;

/**
 * ヘッダーナビゲーション
 */
export const Header: FC<HeaderProps> = (props) => {
  const [isOpen, setIsOpen] = useState(false);
  const handleClick = () => {
    setIsOpen(!isOpen);
  };

  return (
    <StyledComponent {...props} isOpen={isOpen} handleClick={handleClick} />
  );
};
