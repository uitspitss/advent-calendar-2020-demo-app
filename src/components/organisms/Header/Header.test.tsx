import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from './Header';

describe('Header', () => {
  test('Snap Shot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
