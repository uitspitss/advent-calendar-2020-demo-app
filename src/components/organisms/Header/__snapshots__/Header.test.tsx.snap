// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snap Shot 1`] = `
.emotion-4 > div {
  --bg-opacity: 1;
  background-color: rgba(163,191,250,var(--bg-opacity));
  margin-bottom: 0.5rem;
}

@media (min-width:1024px) {
  .emotion-4 > div {
    margin-bottom: 1rem;
  }
}

.emotion-4 > div > nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

@media (min-width:1024px) {
  .emotion-4 > div > nav {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-4 > div > nav > .title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.5rem;
}

.emotion-4 > div > nav > .title > span {
  --text-opacity: 1;
  color: rgba(247,250,252,var(--text-opacity));
  font-weight: 600;
  font-size: 1.25rem;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
}

.emotion-4 > div > nav > .toggleButton {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:1024px) {
  .emotion-4 > div > nav > .toggleButton {
    display: none;
  }
}

.emotion-4 > div > nav > .menuList {
  width: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: none;
}

@media (min-width:1024px) {
  .emotion-4 > div > nav > .menuList {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: auto;
  }
}

@media (min-width:1024px) {
  .emotion-4 > div > nav > .menuList {
    display: block;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  border-width: 0;
  border-radius: 0.25rem;
  background-color: transparent;
}

.emotion-0 svg {
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-0 svg path {
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,-webkit-transform;
  -webkit-transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,transform;
  -webkit-transition-duration: 100ms;
  transition-duration: 100ms;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,1,1);
  transition-timing-function: cubic-bezier(0.4,0,1,1);
}

.emotion-0 svg path.open {
  opacity: 0;
}

.emotion-0 svg path.closed {
  opacity: 1;
}

.emotion-2 {
  font-size: 0.875rem;
}

@media (min-width:1024px) {
  .emotion-2 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

.emotion-2 > a {
  display: block;
  margin-top: 1rem;
  --text-opacity: 1;
  color: rgba(160,174,192,var(--text-opacity));
}

@media (min-width:1024px) {
  .emotion-2 > a {
    display: inline-block;
    margin-top: 0;
  }
}

.emotion-2 > a:not(:last-child) {
  margin-right: 1rem;
}

<header
  className="emotion-4 emotion-5"
>
  <div>
    <nav>
      <div
        className="title"
      >
        <span>
          *WEBSITE TITLE*
        </span>
      </div>
      <div
        className="toggleButton"
      >
        <button
          className="emotion-0 emotion-1"
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className="open"
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
            />
            <path
              className="closed"
              d="M4 6h16M4 12h16M4 18h16"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
            />
          </svg>
        </button>
      </div>
      <div
        className="menuList"
      >
        <div
          className="emotion-2 emotion-3"
        >
          <a
            href="/"
          >
            Docs
          </a>
          <a
            href="/"
          >
            Examples
          </a>
          <a
            href="/"
          >
            Blog
          </a>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
